surveys/tests/test_views_api.py

```diff
@@ -25,3 +28,41 @@ def test_get(self):
             context={'request': request},
         ).data
         self.assertEqual(response.data, expected_data)
+
+    def test_post(self):
+        """Test that the view can retrieve a survey and return correct JSON."""
+        view = views_api.SurveyPostView.as_view()
+        data = {
+            'user_id': 'User#20#',
+            'user_responses': [
+                {
+                    'fieldset': 1,
+                    'answers': {'1': 42}
+                }
+            ],
+        }
+
+        request = self.create_request(method='post', data=data)
+        response = view(request, pk=self.survey.pk)
+
+        self.assertEqual(response.status_code, 201, response.data)
+
+        # Assert we return the same data but with the survey and user_id re-specified
+        # on the objects that need them.
+        data['survey'] = self.survey.pk
+        data['user_responses'][0]['user_id'] = data['user_id']
+        self.assertEqual(response.data, data)
+
+        # Assert the UserResponse object was created correctly.
+        user_response = UserResponse.objects.first()
+        self.assertEqual(user_response.user_id, data['user_id'])
+        self.assertEqual(user_response.answers, data['user_responses'][0]['answers'])
```

Comment:
Three tests?  I can see a way to split it into two but I don't think there's much use in separating out the `user_response` asserts.